tailfs/local.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package tailfs
+
+import (
+	"log"
+	"net"
+	"net/http"
+
+	"github.com/tailscale/xnet/webdav"
+	"tailscale.com/connlistener"
+	"tailscale.com/tailfs/compositefs"
+	"tailscale.com/tailfs/webdavfs"
+	"tailscale.com/types/logger"
+)
+
+// Remote represents a remote TailFS node.
+type Remote struct {
+	Name      string
+	URL       string
+	Available func() bool
+}
+
+// ForLocal is the TailFS filesystem exposed to local clients. It provides a
+// unified WebDAV interface to remote TailFS shares on other nodes.
+type ForLocal interface {
+	// SetRemotes sets the complete set of remotes on the given tailnet domain
+	// using a map of name -> url. If transport is specified, that transport
+	// will be used to connect to these remotes.
+	SetRemotes(domain string, remotes []*Remote, transport http.RoundTripper)
+
+	// HandleConn handles connections from local WebDAV clients
+	HandleConn(conn net.Conn, remoteAddr net.Addr) error
+
+	// Close() stops serving the WebDAV content
+	Close() error
+}
+
+// NewFileSystemForLocal starts serving a filesystem for local clients.
+// Inbound connections must be handed to HandleConn.
+func NewFileSystemForLocal(logf logger.Logf) ForLocal {
```

Comment:
Probably document that `logf` must not be nil, or perhaps give it a default of `logger.Discard`.